base/util.go

```diff
@@ -1039,7 +1039,25 @@ func InjectJSONProperties(b []byte, kvPairs ...KVPair) (new []byte, err error) {
 
 	kvPairsBytes := make([]KVPairBytes, len(kvPairs))
 	for i, kv := range kvPairs {
-		valBytes, err := JSONMarshal(kv.Val)
+		var valBytes []byte
+		var err error
+
+		switch v := kv.Val.(type) {
+		case int, int8, int16, int32, int64:
+			valBytes = []byte(strconv.Itoa(kv.Val.(int)))
+		case uint, uint8, uint16, uint32, uint64:
+			valBytes = []byte(strconv.FormatUint(v.(uint64), 10))
+		case string:
+			valBytes = []byte(`"` + v + `"`)
+		case bool:
+			valBytes = []byte(strconv.FormatBool(v))
+		case float32:
+			valBytes = []byte(fmt.Sprintf("%v", v))
+		case float64:
+			valBytes = []byte(fmt.Sprintf("%v", v))
```

Comment:
For float32 and float64, does fmt.Sprintf("%f") result in the same output as floatEncoder.encode in the standard json library?  I don't think we have any use cases for float injection today, so if we're not sure it's consistent, I'd prefer to leave float unsupported by InjectJSONProperties.

Reply:
'f' doesn't follow the same output but 'v' / 'g' appears to. I will write a couple more tests to confirm it.

Reply:
Ok - have a look at line 559 of https://golang.org/src/encoding/json/encode.go to work out what test values you should include.